go/database/vt/state_test.go

```diff
@@ -0,0 +1,486 @@
+	}
+
+	gethSt, err := geth.NewState(state.Parameters{})
+
+	st.put("memory", memory.NewState())
+	st.put("geth", gethSt)
+
+	return st, err
+}
+
+// comparingState is a state.State implementation that compares multiple
+// state.State instances. It is used to ensure that all operations
+// performed on the state are consistent across all provided states.
+// It implements the state.State interface and provides methods to
+// manipulate accounts, storage, and code in a consistent manner across
+// multiple state instances. The methods defined in this struct will
+// call the corresponding methods on each state instance and return an
+// error if any of the states return an error. This is useful for
+// testing and debugging purposes, where you want to ensure that all
+// state instances behave consistently with each other.
+type comparingState struct {
+	states map[string]state.State // map of state names to state instances
+	t      *testing.T
+}
+
+// put adds a new state to the comparingState instance.
+func (s *comparingState) put(name string, state state.State) {
+	s.states[name] = state
+}
+
+func (s *comparingState) Exists(address common.Address) (bool, error) {
+	s.t.Helper()
+	return getCmpStateValue(s, func(name string, state state.State) (bool, error) {
+		return state.Exists(address)
+	})
+}
+
+func (s *comparingState) GetNonce(address common.Address) (common.Nonce, error) {
+	s.t.Helper()
+	return getCmpStateValue(s, func(name string, state state.State) (common.Nonce, error) {
+		return state.GetNonce(address)
+	})
+}
+
+func (s *comparingState) GetStorage(address common.Address, key common.Key) (common.Value, error) {
+	s.t.Helper()
+	return getCmpStateValue(s, func(name string, state state.State) (common.Value, error) {
+		return state.GetStorage(address, key)
+	})
+}
+
+func (s *comparingState) GetCode(address common.Address) ([]byte, error) {
+	s.t.Helper()
+	b, err := getCmpStateValue(s, func(name string, state state.State) (immutable.Bytes, error) {
+		b, err := state.GetCode(address)
+		return immutable.NewBytes(b), err
+	})
+	return b.ToBytes(), err
+}
+
+func (s *comparingState) GetCodeSize(address common.Address) (int, error) {
+	s.t.Helper()
+	return getCmpStateValue(s, func(name string, state state.State) (int, error) {
+		return state.GetCodeSize(address)
+	})
+}
+
+func (s *comparingState) GetCodeHash(address common.Address) (common.Hash, error) {
+	s.t.Helper()
+	return getCmpStateValue(s, func(name string, state state.State) (common.Hash, error) {
+		return state.GetCodeHash(address)
+	})
+}
+
+func (s *comparingState) GetHash() (common.Hash, error) {
+	s.t.Helper()
+	return getCmpStateValue(s, func(name string, state state.State) (common.Hash, error) {
+		return state.GetHash()
+	})
+}
+
+func (s *comparingState) HasEmptyStorage(addr common.Address) (bool, error) {
+	s.t.Helper()
+	return getCmpStateValue(s, func(name string, state state.State) (bool, error) {
+		return state.HasEmptyStorage(addr)
+	})
+}
```

Comment:
This function is not used in the tests and I think a Verkle Trie might not be able to answer that.